webpack.config.js

```diff
@@ -1,21 +1,25 @@
-var webpack = require('webpack'),
-path = require('path'),
-fileSystem = require('fs-extra'),
-ENV = require('./utils/env'),
-{ CleanWebpackPlugin } = require('clean-webpack-plugin'),
-CopyWebpackPlugin = require('copy-webpack-plugin'),
-HtmlWebpackPlugin = require('html-webpack-plugin'),
-WriteFilePlugin = require('write-file-webpack-plugin'),
-CrxWebpackPlugin = require("crx-webpack-plugin");
+const webpack = require('webpack'),
+  path = require('path'),
+  fileSystem = require('fs-extra'),
+  env = require('./utils/env'),
+  CopyWebpackPlugin = require('copy-webpack-plugin'),
+  HtmlWebpackPlugin = require('html-webpack-plugin'),
+  TerserPlugin = require('terser-webpack-plugin');
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
 
-// load the secrets
-var alias = {
+const ASSET_PATH = process.env.ASSET_PATH || '/';
+
+const alias = {
   'react-dom': '@hot-loader/react-dom',
+  // d3-dsv csvParse require 'unsafe-eval' CSP, which refused by manifest v3
+  // so temporily alias this package to the modified pakage in src
+  'd3-dsv': path.resolve(__dirname, 'src/components/DynamicBar/d3-dsv-2.0.0'),
```

Comment:
DynamicBar uses anichart.js, anichart.js uses d3-dsv, d3-dsv's csvParse require 'unsafe-eval' in Content Security Policy, which refused by manifest v3 extension out of security consideration.

d3-dsv's author is a little subborn, but friends from Open Source world gives help: https://github.com/d3/d3-dsv/pull/80.

I modified d3-dsv source code and place it in our ./src to be part of the project's code, then define an alias in webpack.config.js to tell webpack to search this package in our dir rather node_modules dir.

I know it is not elegant, but it works and when new DynamicBar (absolutely not based on anichart.js) is prepared I will remove these workarounds.